csharp/src/Apache.Arrow/Arrays/Array.cs

```diff
@@ -59,5 +59,22 @@ internal static void Accept<T>(T array, IArrowArrayVisitor visitor)
                     break;
             }
         }
+
+        public Array Slice(int offset, int length)
+        {
+            if (offset > Length)
+            {
+                throw new ArgumentException($"Offset {offset} cannot be greater than Length {Length} for Array.Slice");
+            }
+
+            length = Math.Min(Data.Length - offset, length);
+            offset += Data.Offset;
+
+            ArrayData newData = new ArrayData(Data);
+            newData.Length = length;
+            newData.Offset = offset;
+
+            return ArrowArrayFactory.BuildArray(newData) as Array;
```

Comment:
@chutchinson - what's the use case for having both `IArrowArray` and the abstract base class `Array`? What if we just had the abstract base class and didn't have the interface?